tensorflow/core/kernels/mkl_softmax_op_test.cc

```diff
@@ -1,46 +1,197 @@
+            .Finalize(node_def()));
+    TF_EXPECT_OK(InitOp()); //initial
+    switch(input_type) {
+        case DT_FLOAT:
+            AddInputFromArray<float>(input.shape(), input.flat<float>()); // input
+            break;
+        case DT_BFLOAT16:
+            AddInputFromArray<Eigen::bfloat16>(input.shape(), input.flat<Eigen::bfloat16>()); // input
+            break;
+        default:
+            GTEST_FAIL() << "Unexpected DataType";
+    }
+    AddInputFromArray<uint8_t>(zeros.shape(), zeros.flat<uint8_t>()); // input
+    TF_EXPECT_OK(RunOpKernel()); //Run the node computation
+    mkl_values = *GetOutput(0); //Get output
+    }
+ public:
+    static std::string getTestCaseName(::testing::TestParamInfo<SoftmaxTestParams> obj) {
+        DataType input_type;
+        std::vector<long long int> input_size;
+        std::tie(input_type, input_size) = obj.param;
+        std::ostringstream result;
+        result << "Softmax_Type_";
+        switch(input_type) {
+            case DataType::DT_FLOAT:
+                result << "FLOAT";
+                break;
+            case DataType::DT_BFLOAT16:
+                result << "BFLOAT16";
+                break;
+            default:
+                result << "UNRECOGNISED_TYPE";
+        }
+        result << "_Sizes";
+        for (auto &x : input_size) {
+            result << "_" << x;
+        }
+        return result.str();
+    }
+
+    void SetUp() {
+        std::tie(input_type, input_size) = this->GetParam();
+        input = Tensor(input_type, TensorShape(tensorflow::gtl::ArraySlice<long long int>(input_size.data(), input_size.size())));
+        switch(input_type) {
+            case DT_FLOAT:
+                input.flat<float>() = input.flat<float>().template setRandom<Eigen::internal::NormalRandomGenerator<float>>(); // input
+                break;
+            case DT_BFLOAT16:
+                input.flat<Eigen::bfloat16>() = input.flat<Eigen::bfloat16>().template setRandom<Eigen::internal::UniformRandomGenerator<Eigen::bfloat16>>(); // input
+		        input.flat<Eigen::bfloat16>() = input.flat<Eigen::bfloat16>() - input.flat<Eigen::bfloat16>().constant((Eigen::bfloat16)0.5);
+		        input.flat<Eigen::bfloat16>() = input.flat<Eigen::bfloat16>() * input.flat<Eigen::bfloat16>().constant((Eigen::bfloat16)200.0);
+                break;
+            default:
+                GTEST_FAIL() << "Unexpected DataType";
+        }
+        zeros = Tensor(DT_UINT8, TensorShape({64, 64}));
+        auto zeros_mapped = zeros.tensor<uint8_t, 2>();
+        for(int i = 0; i < 64; i++){
+            for(int j = 0; j < 64; j++){
+                zeros_mapped(i, j) = 0;
+            }
+        }
+    }
+
+    void Run() {
+        runDefault();
+        runMkl();
+    }
+
+    void Validate() {
+        ASSERT_EQ(default_values.dtype(), mkl_values.dtype());
+        ASSERT_EQ(default_values.shape(), mkl_values.shape());
+        test::ExpectClose(default_values, mkl_values, 1e-4);
```

Comment:
Here, can we treat different data type with different data precision? 
such as, we use `1e-7` to check the `float` data type.

Reply:
@jmaksymc

Reply:
Yes, working on it.